src/lucky_router/fragment.cr

```diff
@@ -59,8 +59,30 @@ class LuckyRouter::Fragment(T)
   # the final step to finding the payload is to search for a matching request method
   getter method_to_payload = Hash(String, T).new
 
+  def initialize(@dynamic = false)
+  end
+
+  # This looks for a matching fragment for the given parts
+  # and returns NoMatch if one is not found
   def find(parts : Array(String), method : String) : Match(T) | NoMatch
-    MatchFinder(T).new(self, parts: parts, method: method).run
+    # params are a key value pair of a path variable name matched to its value
+    # so a path like /users/:id will have a path variable name of id and
+    # a matching url of /users/456 will have a value of 456
+    params = {} of String => String
+    result = parts.reduce(self) do |fragment, part|
+      match = fragment.static_parts[part]? || fragment.dynamic_part.try(&.fragment)
+      return NoMatch.new if match.nil?
```

Comment:
Could you refactor this to use implicit returns instead? I find them a bit tricky to work with and refactor since the behavior can change when extacted. For example if this this block of code inside the do/end were moved to a private method the early return would no longer work because it'd return from the new method and not this one.